demoScripts/vmssQuickstart.md

```diff
@@ -0,0 +1,254 @@
+export BACKEND_POOL_NAME=myScaleSetLBBEPool
+export LOAD_BALANCER_RULE_NAME=myLoadBalancerRuleWeb
+export FRONT_END_IP_NAME=loadBalancerFrontEnd 
+export CUSTOM_SCRIPT_NAME=customScript
+export SCALE_SET_PUBLIC_IP=myScaleSetLBPublicIP
+```
+
+## Create a scale set
+Before you can create a scale set, create a resource group with [az group create](/cli/azure/group). The following example creates a resource group named *myResourceGroup* in the *eastus* location:
+
+```azurecli-interactive
+az group create --name $RESOURCE_GROUP_NAME --location $RESOURCE_LOCATION
+```
+
+<!--expected_similarity=0.2-->
+```Output
+{
+  "id": "/subscriptions/<guid>/resourceGroups/myResourceGroup",
+  "location": "eastus",
+  "managedBy": null,
+  "name": "myResourceGroup",
+  "properties": {
+    "provisioningState": "Succeeded"
+  },
+  "tags": null,
+  "type": "Microsoft.Resources/resourceGroups"
+}
+```
+
+Now create a Virtual Machine Scale Set with [az vmss create](/cli/azure/vmss). The following example creates a scale set named *myScaleSet* that is set to automatically update as changes are applied, and generates SSH keys if they do not exist in *~/.ssh/id_rsa*. These SSH keys are used if you need to log in to the VM instances. To use an existing set of SSH keys, instead use the `--ssh-key-value` parameter and specify the location of your keys.
+
+```azurecli-interactive
+az vmss create \
+  --resource-group $RESOURCE_GROUP_NAME \
+  --name $SCALE_SET_NAME \
+  --image $BASE_VM_IMAGE \
+  --upgrade-policy-mode automatic \
+  --admin-username $ADMIN_USERNAME \
+  --generate-ssh-keys
+```
+
+It takes a few minutes to create and configure all the scale set resources and VMs.
+
+
+## Deploy sample application
+To test your scale set, install a basic web application. The Azure Custom Script Extension is used to download and run a script that installs an application on the VM instances. This extension is useful for post deployment configuration, software installation, or any other configuration / management task. For more information, see the [Custom Script Extension overview](../virtual-machines/extensions/custom-script-linux.md).
+
+Use the Custom Script Extension to install a basic NGINX web server. Apply the Custom Script Extension that installs NGINX with [az vmss extension set](/cli/azure/vmss/extension) as follows:
+
+```azurecli-interactive
+az vmss extension set \
+  --publisher Microsoft.Azure.Extensions \
+  --version 2.0 \
+  --name $CUSTOM_SCRIPT_NAME \
+  --resource-group $RESOURCE_GROUP_NAME \
+  --vmss-name $SCALE_SET_NAME \
+  --settings '{"fileUris":["https://raw.githubusercontent.com/Azure-Samples/compute-automation-configurations/master/automate_nginx.sh"],"commandToExecute":"./automate_nginx.sh"}'
+```
+<!--expected_similarity=0.25-->
+```Output
+{
+  "vmss": {
+    "doNotRunExtensionsOnOverprovisionedVMs": false,
+    "orchestrationMode": "Uniform",
+    "overprovision": true,
+    "provisioningState": "Succeeded",
```

Comment:
I wonder if we should consider an issue to enhance the results section. For items like this one that have an explicit result value within them could we add an alternative test using a regular expression? Something like:

```
<-- expected_result=.*\"provisioningState\": \"Succeeded\" -->
```

Not suggesting we block this PR on this item, if it seems like a reasonable idea lets create an issue for it.

As well as being more reliable in the test it makes the doc much more readable as it would become something like:

When this command returns you will get a lot of information in JSON format. The most important thing at this point is that you see "provisioningState": "Succeeded", in the first few lines.

Reply:
I think that is a good idea. The matching algorithm could be improved a lot imo. We could also settle on a specific format for variable outputs like   "id": "/subscriptions/**<guid>**/resourceGroups/myResourceGroup/providers/Microsoft.Compute/virtualMachines/myVM", 

And then take that out of the match portion.

Reply:
I'm not 100% clear on what you mean here. I've created a tracking issue, see #14 . The goal of this issue is simply to get a true/false test result.